delta-app/src/main/java/io/cdap/delta/app/CapacityBoundedEventQueue.java

```diff
@@ -84,13 +81,14 @@ public boolean offer(Sequenced<? extends ChangeEvent> event) {
     lock.lock();
     try {
       int eventSize = computeSize(event);
-      if (size() > 0 && usage + eventSize > capacity) {
+      if (usage + eventSize > capacity) {
```

Comment:
`size() > 0` is needed , it was suggested by Albert for the case when a single event is bigger than the whole queue capacity , we should not block for such event.

Reply:
probably , we can add some comments here so that later other team member can understand it :)

Reply:
sg done!